### PR TITLE
Document dependencies #trivial

### PR DESCRIPTION
**Pull request by djbe**
Saw this blog post and thought it'd be a good idea to document the project's dependencies: http://blog.mazur.me/DEPENDENCIES.md

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: DEPENDENCIES.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- License: [MIT](https://github.com/SwiftGen/StencilSwiftKit/blob/master/LICENCE)
+
+### SwiftLint
+
+- Added to enforce a code style throughout the project, and avoid some bad coding decisions.
+- License: [MIT](https://github.com/realm/SwiftLint/blob/master/LICENSE)
+
+### Yams
+
+- Added because there's no built-in YAML parser, used for configuration file parsing.
```
Comment (AliSoftware) [inline DEPENDENCIES.md]: used for parsing the configuration file and for the YAML parser

**Review by AliSoftware — APPROVED** [review]

**Comment by AliSoftware** [comment]
mmmh wait, I might have approved too quickly 😅 Shouldn't we use the occasion to add a Danger rule that would check if we added (or removed) a Dependency in our Podfile and/or Package.swift, and suggest to update that file in those occasions? Otherwise I fear that we might easily forget to keep that DEPENDENCIES.md up-to-date.

**Comment by djbe** [comment]
You wanna tackle that one? Your ruby-fu is stronger than mine 😉 

Note: options safety PR coming soon.

**Comment by AliSoftware** [comment]
Sure. Going to bed rn though and tomorrow will probably be quite busy, will try and take that on Sunday 👍

**Review by AliSoftware — CHANGES_REQUESTED** [review]
- [x] Danger rule

File: Dangerfile [hunk 2]
```diff
@@ -54,6 +55,13 @@ if podfile_changed ^ package_changed
   need_fixes << warn("You should make sure that `Podfile.lock` and `Package.resolved` are changed in sync")
 end
 
+# Check if DEPENDENCIES needs changes
+podfile_dependencies = YAML::load_file('Podfile.lock')['SPEC CHECKSUMS'].keys.sort
```
Comment (AliSoftware) [inline Dangerfile]: I think `SPEC CHECKSUMS` might contain checksums of _all_ dependencies, including transitive ones.

In our current state, we don't have many transitive dependencies, and the ones we have (mainly `PathKit` coming from `Stencil` and `Stencil` coming from `StencilSwiftKit`), but imagine if `Kanna` had a dependency to `libXML` and `YAMS` had a dependency to hypothetical `Scanner`, `libYAML` and `Colorize` pods… we shouldn't require to document those, as they'd not be dependencies we chose to add directly, but transitive one coming as dependencies from other pods.

What we might want to document though:

* Dependencies explicitly added in our `Podfile` for SwiftGen-the-tool: we could parse those from the `PODS` key of the `Podfile.lock`, trimming away the version in parentheses to only keep the pod name (another idea would be to run `bundle exec pod ipc podfile Podfile` and parse all the values under any `dependencies` keys in that tree, recursively. But not sure that's worth the effort)
* Dependencies listed in `SwiftGenKit.podspec` as SwiftGenKit is also hosted in this repo and is the core of SwiftGen. We can parse those using `bundle exec pod pic spec SwiftGenKit.podspec` and extract the `dependencies` from the resulting JSON output.

**Review by AliSoftware — CHANGES_REQUESTED** [review]

File: Dangerfile (hunk 2 above)
Comment (AliSoftware) [reply Dangerfile]: Also given that the base of those checks are not dependent on GIT and changes made in a PR, but could be run at any time as they compare existing file contents, we might want to move that portion of the code (listing missing dependencies in `DEPENDENCIES.md`) in a `rake` task instead, and in the `Dangerfile` just call that task — but only if the `Podfile.lock` has changed.

**Comment by djbe** [comment]
👍 

Can't approve 'cause it's my own PR 🤦‍♂️

**Review by AliSoftware — APPROVED** [review]

**Pull request merged**